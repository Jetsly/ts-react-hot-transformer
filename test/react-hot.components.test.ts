import * as ts from 'typescript';
import { join } from 'path';
import { readdirSync, readFileSync, statSync } from 'fs';
import transformerFactory from '../src/react-hot.dev';
import transformerProdFactory from '../src/react-hot.prod';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');
function addRHLPlugin(prod = false) {
  return prod ? transformerProdFactory : transformerFactory;
}

describe('React components', () => {
  readdirSync(FIXTURES_DIR).forEach(fixtureName => {
    const fixtureFile = join(FIXTURES_DIR, fixtureName);
    const isForProd = fixtureName.endsWith('.prod.js');
    if (statSync(fixtureFile).isFile()) {
      // if (statSync(fixtureFile).isFile()) {
      it(fixtureName.split('-').join(' '), () => {
        const sourceCode = readFileSync(fixtureFile, 'utf-8');
        const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
        const result = ts.transform(source, [addRHLPlugin(isForProd)]);
        const transformedSourceFile = result.transformed[0];
        const resultCode = printer.printFile(transformedSourceFile);
        expect(resultCode).toMatchSnapshot();
      });
    }
  });
});
