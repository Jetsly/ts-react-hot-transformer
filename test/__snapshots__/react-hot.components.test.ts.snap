// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React components bindings.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
import P, { Q } from \\"left-pad\\";
const A = 42;
function B() {
    function R() { }
    class S {
    }
    const T = 42;
}
export class C {
    U() {
        function V() {
            class W {
            }
        }
    }
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}
const D = class X {
};
let E = D;
var Y = require(\\"left-pad\\");
var { Z } = require(\\"left-pad\\");
const _default = React.createClass({})
export default _default;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(A, \\"A\\", \\"bindings.js\\");
reactHotLoader.register(B, \\"B\\", \\"bindings.js\\");
reactHotLoader.register(C, \\"C\\", \\"bindings.js\\");
reactHotLoader.register(D, \\"D\\", \\"bindings.js\\");
reactHotLoader.register(E, \\"E\\", \\"bindings.js\\");
reactHotLoader.register(_default, \\"default\\", \\"bindings.js\\");

leaveModule(module);
}());


;
"
`;

exports[`React components counter.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Counter {
}
const _default = () => <div><Counter /></div>
export default _default;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Counter, \\"Counter\\", \\"counter.js\\");
reactHotLoader.register(_default, \\"default\\", \\"counter.js\\");

leaveModule(module);
}());


;
"
`;

exports[`React components drop hot.prod.js 1`] = `
"import React from 'react';
import { hot } from 'react-hot-loader/root';
import { hot as namedHot, foo as namedFoo } from 'react-hot-loader/root';
import { hot as namedHot2 } from 'react-hot-loader';
import { hot as notRHLHot } from 'not-react-hot-loader';
import * as RHL from 'react-hot-loader/root';
import * as RHL2 from 'react-hot-loader/root';
import * as NOTRHL from 'not-react-hot-loader';
const App = () => <div>Hello World!</div>;
const a = App;
const b = App;
const c = App;
const d = App;
const e = App;
foo(App);
notRHLHot(App);
namedFoo(App);
RHL.foo(App);
NOTRHL.hot(App);
export { a, b, c, d, e };
"
`;

exports[`React components export default anonymous.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
const _default = function () { }
export default _default;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(_default, \\"default\\", \\"export-default-anonymous.js\\");

leaveModule(module);
}());


;
"
`;

exports[`React components export default class.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
const _default = class a {
}
export default _default;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(_default, \\"default\\", \\"export-default-class.js\\");

leaveModule(module);
}());


;
"
`;

exports[`React components name clash.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
const _default = 10;
const _default2 = 42
export default _default2;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(_default, \\"_default\\", \\"name-clash.js\\");
reactHotLoader.register(_default2, \\"default\\", \\"name-clash.js\\");

leaveModule(module);
}());


;
"
`;
